subprojects/core/src/main/java/org/gradle/internal/service/scopes/GlobalScopeServices.java

```diff
@@ -170,8 +172,12 @@ void configure(ServiceRegistration registration, ClassLoaderRegistry classLoader
         }
     }
 
-    GradleLauncherFactory createGradleLauncherFactory(ListenerManager listenerManager, ProgressLoggerFactory progressLoggerFactory, GradleUserHomeScopeServiceRegistry userHomeScopeServiceRegistry) {
-        return new DefaultGradleLauncherFactory(listenerManager, progressLoggerFactory, userHomeScopeServiceRegistry);
+    StoppableScheduledExecutor createScheduledExecutorService(ExecutorFactory executorFactory) {
```

Comment:
This is generally not the pattern we use for executors. Instead each thing that needs to do something asynchronously would use `ExecutorFactory` to create an executor and then would also take care of stopping this executor. This way, a service does not accidentally leave jobs running when it stops. I would use this pattern for scheduled execution as well.